src/sas/sascalc/shape2sas/Models.py

```diff
@@ -0,0 +1,436 @@
+             + self.z_add * (np.cos(self.gam) * np.sin(self.beta) * np.cos(self.alpha) + np.sin(self.gam) * np.sin(self.alpha)))
+
+        y_rot = (self.x_add * np.sin(self.gam) * np.cos(self.beta)
+             + self.y_add * (np.sin(self.gam) * np.sin(self.beta) * np.sin(self.alpha) + np.cos(self.gam) * np.cos(self.alpha))
+             + self.z_add * (np.sin(self.gam) * np.sin(self.beta) * np.cos(self.alpha) - np.cos(self.gam) * np.sin(self.alpha)))
+
+        z_rot = (-self.x_add * np.sin(self.beta)
+            + self.y_add * np.cos(self.beta) * np.sin(self.alpha)
+            + self.z_add * np.cos(self.beta) * np.cos(self.alpha))
+
+        x_rot += self.rotp_x
+        y_rot += self.rotp_y
+        z_rot += self.rotp_z
+
+        return x_rot, y_rot, z_rot
+
+    #More advanced rotation functions can be added here
+    #but GeneratePoints should be changed....
+
+
+class Translation:
+    def __init__(self, x_add: np.ndarray,
+                       y_add: np.ndarray,
+                       z_add: np.ndarray,
+                       com_x: float,
+                       com_y: float,
+                       com_z: float):
+        self.x_add = x_add
+        self.y_add = y_add
+        self.z_add = z_add
+        self.com_x = com_x
+        self.com_y = com_y
+        self.com_z = com_z
+
+    def onTranslatingPoints(self) -> Vector3D:
+        """Translates points"""
+        return self.x_add + self.com_x, self.y_add + self.com_y, self.z_add + self.com_z
+
+
+class GeneratePoints:
+    def __init__(self, com: list[float],
+                    subunitClass: object,
+                    dimensions: list[float],
+                    rotation: list[float],
+                    rotation_point: list[float],
+                    Npoints: int):
+
+        self.com = com
+        self.subunitClass = subunitClass
+        self.dimensions = dimensions
+        self.rotation = rotation
+        self.rotation_point = rotation_point
+        self.Npoints = Npoints
+
+    def onGeneratingPoints(self) -> Vector3D:
+        """Generates the points"""
+        print(f"Generating {self.Npoints} points for subunit {self.subunitClass.__name__} with dimensions {self.dimensions}")
+        x, y, z= self.subunitClass(self.dimensions).getPointDistribution(self.Npoints)
+        x, y, z = self.onTransformingPoints(x, y, z)
+        return x, y, z
+
+    def onTransformingPoints(self, x: np.ndarray,
+                                   y: np.ndarray,
+                                   z: np.ndarray) -> Vector3D:
+        """Transforms the points"""
+        alpha, beta, gam = self.rotation
+        rotp_x, rotp_y, rotp_z = self.rotation_point
+        alpha = np.radians(alpha)
+        beta = np.radians(beta)
+        gam = np.radians(gam)
+        com_x, com_y, com_z = self.com
+
+        x, y, z = Rotation(x, y, z, alpha, beta, gam, rotp_x, rotp_y, rotp_z).onRotatingPoints()
+        x, y, z = Translation(x, y, z, com_x, com_y, com_z).onTranslatingPoints()
```

Comment:
Your Rotation and Translation classes would be better written as functions. You create them, call them immediately, then through them away.